src/device/device_ledger.cpp

```diff
@@ -1056,6 +1045,19 @@ namespace hw {
         hw::ledger::log_hexbuffer("generate_key_derivation: [[OUT]] derivation", derivation_x.data, 32);
         #endif
 
+      if ((this->mode == TRANSACTION_PARSE)  && has_view_key) {
+        MDEBUG( "generate_key_derivation  : PARSE mode with known viewkey");     
+        if (is_fake_view_key(sec)) {
+          r = crypto::generate_key_derivation(pub, this->viewkey, derivation);
+        }
+        else {
+          r = crypto::generate_key_derivation(pub, sec, derivation);
```

Comment:
This branch logic seems wrong; `hwdev.generate_key_derivation(pub, sec, deriv)` is always called with `sec` being the view secret key which is always set to zero, except for the L393 of cryptonote_tx_utils.cpp where `sec` is the tx secret key where decryption is necessary. The correct branch logic IMO is:

```c++
if ((this->mode == TRANSACTION_PARSE)  && has_view_key && is_fake_view_key(sec)) {
  r = crypto::generate_key_derivation(pub, this->viewkey, derivation);
} else {
  ... decryption ...
}
```

Reply:
I mimic the generate_key_derivation general function, which I thinh the best think to do to anticipate any future evolution. So The logic is right regard this anticipation, And i would to keep it :)